src/efftester.ml

```diff
@@ -185,11 +201,26 @@ let term_to_ocaml ?(typeannot = true) term =
       else Printf.bprintf sb "%s" x
     in
     match t with
+    | Constructor (_, name, trm_opt) ->
+      (match trm_opt with
+      | Some trm -> Printf.bprintf sb "(%s (%a))" name exp trm
+      | None -> Printf.bprintf sb "%s" name)
+    | PatternMatch (_, match_trm, branches, _) ->
+      Printf.bprintf
+        sb
+        "match %a with %a"
+        exp
+        match_trm
+        (fun sb lst ->
+          List.iter
+            (fun (case, body) -> Printf.bprintf sb "| %a -> (%a) " exp case exp body)
+            lst)
+        branches
```

Comment:
I think that this code is fine, and the placement of parentheses seems correct to me.

(This overall printing function will generate more parentheses than necessary in some cases, but this is perfectly fine, given that the implementations will just ignore them.)